### PR TITLE
Drivetrain subsystem, Suction subsystem

### PR DESCRIPTION
**Pull request by MeechaPooch**

### CODE REVIEW

File: .run/Monitor Output of Simulate Java.run.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<component name="ProjectRunConfigurationManager">
```
Comment (rutmanz) [inline .run/Monitor Output of Simulate Java.run.xml]: `.run` looks like a bunch of generated files, probably shouldn't be committed

File: src/main/java/org/team1540/elmo/Constants.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    public static final int LEFT_BACK_MOTOR_CAN_ID = 1;
+    public static final int RIGHT_FRONT_MOTOR_CAN_ID = 2;
+    public static final int RIGHT_BACK_MOTOR_CAN_ID = 3;
+
+    // SOLENOIDS
+    public static final int ARM_FORWARD_CHANNEL = 0;
+    public static final int ARM_REVERSE_CHANNEL = 1;
+    public static final PneumaticsModuleType ARM_TYPE = PneumaticsModuleType.CTREPCM;
+    public static final int SUCTION_CHANNEL = 2;
+    public static final PneumaticsModuleType SUCTION_TYPE = PneumaticsModuleType.CTREPCM;
```
Comment (rutmanz) [inline src/main/java/org/team1540/elmo/Constants.java]: I believe there will be a single pneumatics controller, which will be a CTREPCM, it's not really a problem, but you only really need to be declaring that once

File: notes [hunk 3]
```diff
@@ -0,0 +1,2 @@
+install vendor url:
```
Comment (rutmanz) [inline notes]: Might want to use a more descriptive filename/add an extension (like `TODO.md` or `NOTES.md`), just to make it clear what this file is just from the name

File: src/main/java/org/team1540/elmo/drivetrain/DriveTrain.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.team1540.elmo.Constants;
+
+public class DriveTrain extends SubsystemBase {
+
+    TalonFX motorRightFront = new TalonFX(Constants.RIGHT_FRONT_MOTOR_CAN_ID);
+    TalonFX motorRightBack = new TalonFX(Constants.RIGHT_BACK_MOTOR_CAN_ID);
+    TalonFX motorLeftFront = new TalonFX(Constants.LEFT_FRONT_MOTOR_CAN_ID);
+    TalonFX motorLeftBack = new TalonFX(Constants.LEFT_BACK_MOTOR_CAN_ID);
```
Comment (rutmanz) [inline src/main/java/org/team1540/elmo/drivetrain/DriveTrain.java]: I don't think that you will have Falcons on this robot. I believe you'll be using Victor SPX speed controllers for brushed motors, make sure to confirm with Controls/Fab though. The SPXes will still be part of the Phoenix library, so the interface should be pretty similar to the Talons

File: vendordeps/REVLib.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
+{
+    "fileName": "REVLib.json",
```
Comment (rutmanz) [inline vendordeps/REVLib.json]: Do you have any Rev products that will be on your robot? I didn't see the rev stuff used anywhere

**Review by rutmanz — COMMENTED** [review]
There are a few smaller things that could be changed, most important of which being the speed controller class, but all of the big things look pretty good

File: src/main/java/org/team1540/elmo/Constants.java (hunk 2 above)
Comment (MeechaPooch) [reply src/main/java/org/team1540/elmo/Constants.java]: yessir thanks https://github.com/flamingchickens1540/bunnybots-elmo-2022/blob/3b905b8bd5fc935b66e04e30b125a4211ac73cbb/src/main/java/org/team1540/elmo/Constants.java#L28

File: notes (hunk 3 above)
Comment (MeechaPooch) [reply notes]: done https://github.com/flamingchickens1540/bunnybots-elmo-2022/blob/micah/USEFUL_NOTES.md

File: src/main/java/org/team1540/elmo/drivetrain/DriveTrain.java (hunk 4 above)
Comment (MeechaPooch) [reply src/main/java/org/team1540/elmo/drivetrain/DriveTrain.java]: ive removed the drivetrain entirely from this branch

File: vendordeps/REVLib.json (hunk 5 above)
Comment (MeechaPooch) [reply vendordeps/REVLib.json]: gone, reduced to atoms D;;;;; up thanos butt

**Pull request closed without merging**